projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts

```diff
@@ -1,94 +1,147 @@
-import { requestAPI } from "./handler";
- /**
-   * Utility functions that are called by the dialogs.
-   * They are called when a request to the server is needed to be sent.
-   */
+import { requestAPI } from './handler';
+import { Dialog, showErrorMessage } from '@jupyterlab/apputils';
+/**
+ * Utility functions that are called by the dialogs.
+ * They are called when a request to the server is needed to be sent.
+ */
 
-
- /**
-    * Sent a GET request to the server to get current working directory
-    */
-export function getCurrentPathRequest(){
-    requestAPI<any>('run', {
-        method: 'GET',
+/**
+ * Sent a GET request to the server to get current working directory
+ */
+export function getCurrentPathRequest() {
```

Comment:
That looks like a race condition to me. 
I think you need to mark this as async and await the result